Source/Android/jni/MainAndroid.cpp

```diff
@@ -152,8 +152,10 @@ void Host_TitleChanged()
 {
   s_game_metadata_is_valid = true;
 
-  JNIEnv* env = IDCache::GetEnvForThread();
-  env->CallStaticVoidMethod(IDCache::GetNativeLibraryClass(), IDCache::GetOnTitleChanged());
+  std::thread([] {
```

Comment:
This should have a comment explaining why this is required.

Reply:
The problem is that I have no idea why this is required. It's the one thing keeping this PR from being mergeable in my opinion.

Reply:
I had a similar issue and know where it crashes. The culprit is JVM's `AttachCurrentThread` in `GetEnvForThread`.

In IDCache.cpp:
```c++
if (status == JNI_EDETACHED)
  s_java_vm->AttachCurrentThread(&env, nullptr);
```

The things is, for me at least, it raises randomly many different type of exceptions/errors such as segfault, libc/stack corruption, memory access errors, etc. I tried to find any relevant pieces of documentation regarding this behaviour but couldn't find any. I also don't know (yet) how to debug native code so I couldn't go deeper.

I suspect there is something fishy in that function and that depending on how the thread was created it might produce an undefined behaviour.

Reply:
Hmm, so `AttachCurrentThread` fails if we call it from the CPU thread, but only when fastmem is enabled... Maybe we could make sure to attach the CPU thread before enabling fastmem? I'll give that a try.

Reply:
That seems to do the trick. I'm much happier with this than the earlier workaround.

My new commit calls `AttachCurrentThread` before `EMM::InstallExceptionHandler`, but calling it immediately afterwards also seems to work. Perhaps the problem only occurs after the first fastmem fault rather than after the exception handler is installed?